sympy/solvers/solvers.py

```diff
@@ -350,7 +350,11 @@ def check_assumptions(expr, against=None, **assumptions):
         if test is expected:
             continue
         elif test is not None:
-            return False
+            if get is True:
+                return False, key
```

Comment:
For this 

```
check_assumptions(-5, integer=False, positive=True)
```

it must not return only `False, integer `, right ? Also write test cases.